master/buildbot/steps/source/git.py

```diff
@@ -752,8 +758,19 @@ def _doCommit(self):
         for message in self.messages:
             cmd.extend(['-m', message])
 
-        ret = yield self._dovccmd(cmd)
-        return ret
+        if self.emptyCommits == 'create-empty-commit':
+            cmd.extend(['--allow-empty'])
+
+        if self.emptyCommits == 'ignore':
+            stdout, rc = yield self._dovccmd(cmd, collectStdoutAndRC=True, decodeRC={0: SUCCESS, 1: SUCCESS})
+            if rc == 0:
+                return 0
```

Comment:
Looks like there's no unit test covering this code path.

Reply:
Thanks! Added a test for this one.